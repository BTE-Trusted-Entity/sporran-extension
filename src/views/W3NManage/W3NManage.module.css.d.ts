export const __esModule: true;
export const container: string;
export const heading: string;
export const didLine: string;
export const web3NameLine: string;
export const remove: string;
