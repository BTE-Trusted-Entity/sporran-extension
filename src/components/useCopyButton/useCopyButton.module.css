.copy {
  composes: buttonIcon from '../Button/Button.module.css';
  background-image: url('../images/copy.svg');
  background-position: center 40%;
  background-size: auto 50%;
  width: 2.25rem;
  height: 2.25rem;
  position: absolute;
  right: 0.25rem;
}

.copied {
  composes: copy;
  background-image: url('../images/checkmark-circle.svg');
}
