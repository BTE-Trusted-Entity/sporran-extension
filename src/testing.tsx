export * from '@testing-library/react';
export { waitForElementToBeRemoved } from '@testing-library/dom';
import { act, render as externalRender, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import dialogPolyfill from 'dialog-polyfill';

import { AccountsProviderMock } from './testing/AccountsProviderMock';

export {
  AccountsProviderMock,
  accountsMock,
} from './testing/AccountsProviderMock';

export { mockBackgroundScript } from './testing/mockBackgroundScript';

export function render(
  ui: Parameters<typeof externalRender>[0],
  options?: Parameters<typeof externalRender>[1],
): ReturnType<typeof externalRender> {
  return externalRender(
    <AccountsProviderMock>
      <MemoryRouter>{ui}</MemoryRouter>
    </AccountsProviderMock>,
    options,
  );
}

export async function waitForTooltipUpdate(): Promise<void> {
  expect(await screen.findByRole('status')).toHaveAttribute(
    'data-popper-escaped',
  );
}

const dialogPromise = Promise.resolve();
(dialogPolyfill.registerDialog as jest.Mock).mockReturnValue(dialogPromise);

export async function waitForDialogUpdate(): Promise<void> {
  await act(() => dialogPromise);
}
