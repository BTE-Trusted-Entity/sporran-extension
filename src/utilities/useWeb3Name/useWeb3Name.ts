import { ConfigService, Did, DidUri } from '@kiltprotocol/sdk-js';

import { isFullDid } from '../did/did';
import { useAsyncValue } from '../useAsyncValue/useAsyncValue';

export function useWeb3Name(
  did: DidUri | undefined,
): string | null | undefined {
  return useAsyncValue(
    async (did) => {
      if (!(did && isFullDid(did))) {
        return null;
      }

      const api = ConfigService.get('api');
      const result = await api.call.did.query(Did.toChain(did));
      if (result.isNone) {
        return null;
      }

      return Did.linkedInfoFromChain(result).web3Name;
    },
    [did],
  );
}
