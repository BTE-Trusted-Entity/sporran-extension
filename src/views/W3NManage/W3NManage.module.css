.container {
  composes: container from '../../components/View/View.module.css';
}

.heading {
  composes: screenHeadline from '../../components/Typography/Typography.module.css';
  margin-bottom: calc(31rem / 16);
}

.didLine {
  composes: didLine from '../../components/CopyValue/CopyValue.module.css';
  margin-top: 1.5rem;
}

.web3NameLine {
  composes: input from '../../components/CopyValue/CopyValue.module.css';
  text-align: center;
  margin-top: 0.75rem;
  margin-bottom: 0.25rem;

  --copy-value-icon-size: 1.5rem;
  --copy-value-icon: url('../../components/images/web3name.svg');
}

.remove {
  composes: buttonWide from '../../components/Button/Button.module.css';
  margin: 0.75rem 0 0;

  --wide-button-icon: url('../../components/images/web3name.svg');
}
