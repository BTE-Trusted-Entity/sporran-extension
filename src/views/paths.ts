export { generatePath } from 'react-router-dom';

export const paths = {
  home: '/',
  settings: '/settings',
  access: '/access',
  identity: {
    base: '/identity/',
    add: '/identity/add',
    create: {
      start: '/identity/create',
      backup: '/identity/create/backup',
      verify: '/identity/create/verify',
      password: '/identity/create/password',
    },
    import: {
      start: '/identity/import',
      password: '/identity/import/password',
    },

    overview: '/identity/:address/:type(created|imported|pwreset)?',
    reset: {
      start: '/identity/:address/reset',
      password: '/identity/:address/reset/password',
    },
    send: {
      start: '/identity/:address/send',
      review: '/identity/:address/send/review',
      warning: '/identity/:address/send/warning',
    },
    receive: '/identity/:address/receive',
    credentials: {
      base: '/identity/:address/credentials',
      presentation: '/identity/:address/credentials/:hash/presentation',
    },
    remove: '/identity/:address/remove',
    did: {
      upgrade: {
        start: '/identity/:address/did/upgrade',
        kilt: '/identity/:address/did/upgrade/kilt',
        euro: '/identity/:address/did/upgrade/euro',
      },
      manage: {
        start: '/identity/:address/did',
        downgrade: {
          base: '/identity/:address/did/downgrade',
          warning: {
            web3name: '/identity/:address/did/downgrade/warning/web3name',
            credentials: '/identity/:address/did/downgrade/warning/credentials',
          },
          sign: '/identity/:address/did/downgrade/sign',
        },
        endpoints: {
          start: '/identity/:address/did/endpoints',
          edit: '/identity/:address/did/endpoints/:id?',
          sign: '/identity/:address/did/endpoints/sign',
        },
      },
      repair: '/identity/:address/did/repair',
    },
    web3name: {
      create: {
        base: '/identity/:address/web3name/create',
        info: '/identity/:address/web3name/create/info',
        form: '/identity/:address/web3name/create/form',
        choose: '/identity/:address/web3name/create/choose/:web3name',
        kilt: '/identity/:address/web3name/create/kilt/:web3name',
        euro: '/identity/:address/web3name/create/euro/:web3name',
      },
      manage: {
        start: '/identity/:address/web3name',
        remove: '/identity/:address/web3name/remove',
      },
    },
  },
  popup: {
    base: '/popup',
    import: '/popup/import',
    access: '/popup/access',
    claim: '/identity/:address/claim',
    save: '/popup/save',
    share: {
      start: '/popup/share',
      sign: '/popup/share/sign',
    },
    sign: '/popup/sign',
    signRaw: '/popup/signRaw',
    signDid: {
      start: '/identity/:address/did/sign',
      sign: '/identity/:address/did/sign/password',
      credentials: '/identity/:address/did/sign/credentials',
    },
    signDidExtrinsic: '/identity/:address/did/signExtrinsic',
    createDid: '/identity/:address/did/create',
  },
};
