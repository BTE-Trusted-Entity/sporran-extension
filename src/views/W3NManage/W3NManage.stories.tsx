import { MemoryRouter, Route } from 'react-router-dom';
import { Meta } from '@storybook/react';

import { moreIdentitiesMock as identities } from '../../utilities/identities/IdentitiesProvider.mock';
import { generatePath, paths } from '../paths';

import { W3NManage } from './W3NManage';

export default {
  title: 'Views/W3NManage',
  component: W3NManage,
} as Meta;

export function Template(): JSX.Element {
  return (
    <MemoryRouter
      initialEntries={[
        generatePath(paths.identity.did.manage.start, { address: 'FOO' }),
      ]}
    >
      <Route path={paths.identity.did.manage.start}>
        <W3NManage
          identity={
            identities['4q11Jce9wqM4A9GPB2z8n4K8LF9w2sQgZKFddhuKXwQ2Qo4q']
          }
        />
      </Route>
    </MemoryRouter>
  );
}
