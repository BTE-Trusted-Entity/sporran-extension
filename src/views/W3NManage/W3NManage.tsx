import { Link } from 'react-router-dom';
import { browser } from 'webextension-polyfill-ts';

import * as styles from './W3NManage.module.css';

import { LinkBack } from '../../components/LinkBack/LinkBack';
import { Stats } from '../../components/Stats/Stats';
import { Identity } from '../../utilities/identities/types';
import { getIdentityDid } from '../../utilities/identities/identities';
import { IdentitySlide } from '../../components/IdentitySlide/IdentitySlide';
import { CopyValue } from '../../components/CopyValue/CopyValue';
import { generatePath, paths } from '../paths';
import { useWeb3Name } from '../../utilities/useWeb3Name/useWeb3Name';

interface Props {
  identity: Identity;
}

export function W3NManage({ identity }: Props): JSX.Element {
  const t = browser.i18n.getMessage;

  const { address } = identity;
  const did = getIdentityDid(identity);

  const web3name = useWeb3Name(did);

  return (
    <section className={styles.container}>
      <h1 className={styles.heading}>{t('view_W3NManage_heading')}</h1>

      <IdentitySlide identity={identity} />

      <CopyValue value={did} label="DID" className={styles.didLine} />

      {web3name && (
        <input
          value={web3name}
          aria-label="web3name"
          readOnly
          className={styles.web3NameLine}
        />
      )}

      {web3name && (
        <Link
          className={styles.remove}
          to={generatePath(paths.identity.web3name.manage.remove, { address })}
        >
          {t('view_W3NManage_remove')}
        </Link>
      )}

      <LinkBack />
      <Stats />
    </section>
  );
}
