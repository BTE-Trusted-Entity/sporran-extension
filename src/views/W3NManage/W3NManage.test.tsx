import { MemoryRouter, Route } from 'react-router-dom';

import { identitiesMock as identities, render } from '../../testing/testing';

import '../../components/useCopyButton/useCopyButton.mock';

import { generatePath, paths } from '../paths';

import { useWeb3Name } from '../../utilities/useWeb3Name/useWeb3Name';

import { W3NManage } from './W3NManage';

jest.mock('../../utilities/useWeb3Name/useWeb3Name');

describe('W3NManage', () => {
  it('should render', async () => {
    jest.mocked(useWeb3Name).mockReturnValue('fancy-name');
    const { container } = render(
      <MemoryRouter
        initialEntries={[
          generatePath(paths.identity.web3name.manage.start, {
            address: 'FOO',
          }),
        ]}
      >
        <Route path={paths.identity.web3name.manage.start}>
          <W3NManage
            identity={
              identities['4pNXuxPWhMxhRctgB4qd3MkRt2Sxp7Y7sxrApVCVXCEcdQMo']
            }
          />
        </Route>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
