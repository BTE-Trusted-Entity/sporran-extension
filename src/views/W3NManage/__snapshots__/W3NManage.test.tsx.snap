// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`W3NManage should render 1`] = `
<div>
  <section
    class="container"
  >
    <h1
      class="heading"
    >
      Manage web3name
    </h1>
    <section
      class="container"
    >
      <div
        class="avatar"
      >
        Identicon
      </div>
      <div
        class="centeredNameLine"
      >
        <span
          class="name"
        >
          Identity with on-chain DID and lots of coins
        </span>
      </div>
    </section>
    <p
      class="didLine"
    >
      <input
        aria-label="DID"
        class="long"
        readonly=""
        value="did:kilt:4oeJ76hdj84xnwCNqijUHUCTmfwXgSZ4vmxLEiTEYgQdBCcZ"
      />
      <button
        aria-label="copy to clipboard"
        class="copy"
        title="copy to clipboard"
        type="button"
      />
    </p>
    <input
      aria-label="web3name"
      class="web3NameLine"
      readonly=""
      value="fancy-name"
    />
    <a
      class="remove"
      href="/identity/4pNXuxPWhMxhRctgB4qd3MkRt2Sxp7Y7sxrApVCVXCEcdQMo/web3name/remove"
    >
      Remove web3name
    </a>
    <button
      aria-label="Back"
      class="linkBack"
      title="Back"
      type="button"
    />
    <p
      class="stats"
    >
      You have 3 Identities with a
      <span
        class="balance"
      >
         Total Balance: 
        <span
          class="amount"
        >
          3.7020
           
          <span
            aria-label="KILT Coin"
            class="coin"
          />
        </span>
      </span>
    </p>
  </section>
</div>
`;
